mesonbuild/backend/ninjabackend.py

```diff
@@ -105,18 +107,17 @@ def write(self, outfile):
             (name, elems) = e
             should_quote = name not in raw_names
             line = ' %s = ' % name
-            q_templ = quote_char + "%s" + quote_char
             noq_templ = "%s"
             newelems = []
             for i in elems:
                 if not should_quote or i == '&&': # Hackety hack hack
-                    templ = noq_templ
+                    quoter = ninja_quote
                 else:
-                    templ = q_templ
+                    quoter = lambda x: ninja_quote(quote_func(x))
```

Comment:
__[flake8]__
* [E731] do not assign a lambda expression, use a def

<!-- Comment by [SideCI](https://sideci.com) -->